### PR TITLE
Conversion implementation

### LINKED ISSUE #278: Provide method to see USD conversion of Eth
Submitted by @eshon 

Need to decide how to show this (design).

Also need to decide what to show on non-mainnet (morden testnet, test blockchains, etc).

### LINKED ISSUE #362: Indicating USD value on tx notifications
Blocked by #278

### LINKED ISSUE #461: Add JPY conversion

### PR DESCRIPTION
**Pull request by 2-am-zzz**
Work in Progress. Fixes #278, Fixes #461, Fixes #362. Pulling a PR here for accountability and team checks.

Essentially does these things (or hopes to do so):
1) Uses the Cryptonator API to pull data occasionally on some fiat currency to ether conversion. Thankfully they don't have limits, so we can ping liberally.

2) Allows users to select from a currently pre-defined list of fiat currencies from the config menu.

3) Shows users the equivalent fiat value when hovering over any ether value.

### CODE REVIEW

**Comment by Zanibas** [comment]
screenshot from 2016-08-16 14 24 14

screenshot from 2016-08-16 14 24 32

screenshot from 2016-08-16 14 24 56

**Comment by Zanibas** [comment]
This PR is finished! Waiting for additional comments or a merge @flyswatter @kumavis @frankiebee

**Comment by kumavis** [comment]
@Zanibas is the NaN issue resolved? (see tx history in screenshots )

**Comment by flyswatter** [comment]
I don't think you need an `=` before the `{value} USD`.

I would also add the currency suffix to the accounts page, and remove the equal sign there too.

**Comment by Zanibas** [comment]
@flyswatter it was suggested in Vlad's design: is it no good?

**Comment by kumavis** [comment]
@Zanibas I don't really like it, maybe use parens instead
```
100 ether
( 1000 USD )
```

**Comment by flyswatter** [comment]
I don't think equals or parens are needed. Different font color + currency suffix will make it very obvious.

File: ui/app/components/account-eth-balance.js [hunk 1]
```diff
@@ -44,13 +44,22 @@ EthBalanceComponent.prototype.render = function () {
   )
 }
 EthBalanceComponent.prototype.renderBalance = function (value, state) {
+  console.log("THIS IS VALUE")
+  console.log(value)
+  console.log(state.conversionRate)
```
Comment (flyswatter) [inline ui/app/components/account-eth-balance.js]: Remove unused logs

File: ui/app/components/account-eth-balance.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        },
+      }, this.renderBalance(value, state)),
+    ])
+
+  )
+}
+EthBalanceComponent.prototype.renderBalance = function (value, state) {
+  console.log("THIS IS VALUE")
+  console.log(value)
+  console.log(state.conversionRate)
```
Comment (flyswatter) [inline ui/app/components/account-eth-balance.js]: Remove unused logs

**Comment by flyswatter** [comment]
Tests are failing. probably linter issues.

This PR looks good!  Great work, @Zanibas, this was a large feature, and you've done it very cleanly, nice tests, good use of our state tree, metamask controller, config manager, very solid contribution!

**Comment by Zanibas** [comment]
Removed the equal sign, but now to account for failing APIs, we just hide that part of the UI (no conversion is shown at all). Showing a message indicating unavailable conversions was pretty ugly--we can revisit this if this is undesired.

**Comment by flyswatter** [comment]
I like the solution of hiding the value when it's unavailable.

**Pull request merged**